packages/frontend-shared/src/components/ResultCounts.vue

```diff
@@ -36,33 +41,50 @@ const { t } = useI18n()
 
 const props = defineProps<ResultCountsProps>()
 
+const ORDER_MAP = {
+  'SKIPPED': {
+    value: props.totalSkipped,
+    class: 'icon-dark-gray-400',
+    icon: SkippedIcon,
+    name: t('runs.results.skipped'),
+  },
+  'PENDING': {
+    value: props.totalPending,
+    class: 'icon-dark-gray-400 icon-light-white',
+    icon: PendingIcon,
+    name: t('runs.results.pending'),
+  },
+  'PASSED': {
+    value: props.totalPassed,
+    class: 'icon-dark-jade-400',
+    icon: PassedIcon,
+    name: t('runs.results.passed'),
+  },
+  'FAILED': {
+    value: props.totalFailed,
+    class: 'icon-dark-red-400',
+    icon: FailedIcon,
+    name: t('runs.results.failed'),
+  },
+}
+
+interface status {
+  value: number|string|null
+  class: string
+  icon: FunctionalComponent<SVGAttributes, {}>
+  name: string
+}
+
 const results = computed(() => {
-  return [
-    {
-      value: props.totalSkipped,
-      class: 'icon-dark-gray-400',
-      icon: SkippedIcon,
-      name: t('runs.results.skipped'),
-    },
-    {
-      value: props.totalPending,
-      class: 'icon-dark-gray-400 icon-light-white',
-      icon: PendingIcon,
-      name: t('runs.results.pending'),
-    },
-    {
-      value: props.totalPassed,
-      class: 'icon-dark-jade-400',
-      icon: PassedIcon,
-      name: t('runs.results.passed'),
-    },
-    {
-      value: props.totalFailed,
-      class: 'icon-dark-red-400',
-      icon: FailedIcon,
-      name: t('runs.results.failed'),
-    },
-  ]
+  const acc: status[] = []
+
+  if (props.order) {
+    props.order.forEach((status) => acc.push(ORDER_MAP[status]))
+  } else {
+    acc.push(ORDER_MAP['SKIPPED'], ORDER_MAP['PENDING'], ORDER_MAP['PASSED'], ORDER_MAP['FAILED'])
+  }
```

Comment:
I always try to think if there are ways to simplify conditionals in a functional way.  I also look at the different parts the code is trying to accomplish.  Here, we need to figure out what order to use first, then use that order to create an array.  

```suggestion
  const order = props.order || ['SKIPPED', 'PENDING', 'PASSED', 'FAILED']

  return order.map((status) => ORDER_MAP[status])
```

Reply:
Also, the Vue `computed` method creates a reactive `ref` that would be updated if any reactive elements inside the `computed` changes.  In order for this to see any changes to the `prop` values for the counts, that map creation needs to move back inside the `computed` method.  Happy to explain more if it would help.